homeassistant/components/here_travel_time/sensor.py

```diff
@@ -90,32 +93,49 @@
 
 NO_ROUTE_ERROR_MESSAGE = "HERE could not find a route based on the input"
 
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Inclusive(
+            CONF_DESTINATION_LATITUDE, "destination_coordinates"
+        ): cv.latitude,
+        vol.Inclusive(
+            CONF_DESTINATION_LONGITUDE, "destination_coordinates"
+        ): cv.longitude,
+        vol.Exclusive(CONF_DESTINATION_LATITUDE, "destination"): cv.latitude,
+        vol.Exclusive(CONF_DESTINATION_ENTITY_ID, "destination"): cv.entity_id,
+        vol.Inclusive(CONF_ORIGIN_LATITUDE, "origin_coordinates"): cv.latitude,
+        vol.Inclusive(CONF_ORIGIN_LONGITUDE, "origin_coordinates"): cv.longitude,
+        vol.Exclusive(CONF_ORIGIN_LATITUDE, "origin"): cv.latitude,
+        vol.Exclusive(CONF_ORIGIN_ENTITY_ID, "origin"): cv.entity_id,
+        vol.Optional(CONF_DEPARTURE): cv.time,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MODE, default=TRAVEL_MODE_CAR): vol.In(TRAVEL_MODE),
+        vol.Optional(CONF_ROUTE_MODE, default=ROUTE_MODE_FASTEST): vol.In(ROUTE_MODE),
+        vol.Optional(CONF_TRAFFIC_MODE, default=False): cv.boolean,
+        vol.Optional(CONF_UNIT_SYSTEM): vol.In(UNITS),
+    }
+)
+
 PLATFORM_SCHEMA = vol.All(
     cv.has_at_least_one_key(CONF_DESTINATION_LATITUDE, CONF_DESTINATION_ENTITY_ID),
     cv.has_at_least_one_key(CONF_ORIGIN_LATITUDE, CONF_ORIGIN_ENTITY_ID),
-    PLATFORM_SCHEMA.extend(
+    cv.key_value_schemas(
+        CONF_MODE,
         {
-            vol.Required(CONF_API_KEY): cv.string,
-            vol.Inclusive(
-                CONF_DESTINATION_LATITUDE, "destination_coordinates"
-            ): cv.latitude,
-            vol.Inclusive(
-                CONF_DESTINATION_LONGITUDE, "destination_coordinates"
-            ): cv.longitude,
-            vol.Exclusive(CONF_DESTINATION_LATITUDE, "destination"): cv.latitude,
-            vol.Exclusive(CONF_DESTINATION_ENTITY_ID, "destination"): cv.entity_id,
-            vol.Inclusive(CONF_ORIGIN_LATITUDE, "origin_coordinates"): cv.latitude,
-            vol.Inclusive(CONF_ORIGIN_LONGITUDE, "origin_coordinates"): cv.longitude,
-            vol.Exclusive(CONF_ORIGIN_LATITUDE, "origin"): cv.latitude,
-            vol.Exclusive(CONF_ORIGIN_ENTITY_ID, "origin"): cv.entity_id,
-            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-            vol.Optional(CONF_MODE, default=TRAVEL_MODE_CAR): vol.In(TRAVEL_MODE),
-            vol.Optional(CONF_ROUTE_MODE, default=ROUTE_MODE_FASTEST): vol.In(
-                ROUTE_MODE
+            None: PLATFORM_SCHEMA,
```

Comment:
This should not be possible since there is a default and `None` is not part of `TRAVEL_MODE` constant.

Reply:
The default is not recognized by `cv.has_at_least_one_key`.

If I leave the line out I get the following error messages for configurations where `CONF_MODE` is not configured and thus the default is used:
```
Unexpected value for mode: 'None'. Expected bicycle, car, pedestrian, publicTransport, truck, publicTransportTimeTable. Got {'platform': 'here_travel_time', 'name': 'test', 'origin_latitude': '41.9798', 'origin_longitude': '-87.8801', 'destination_latitude': '41.9043', 'destination_longitude': '-87.9216', 'api_key': 'test', 'arrival': '01:00:00'}
```

Reply:
Oh, right, the default is only applied once it has picked a schema. I wonder if we should instead allow passing in a default to `key_value_schemas`.

Reply:
That would be helpful. Another thought is that I had to list all possible values for `mode` even though I was only looking for one specific vs all the rest.
I can imagine that gets a hassle for more options.